### ISSUE TITLE
‼️ NOTICE: @types/prettier@2.6.1 breaks builds

### ISSUE THREAD

**Issue by corymhall** [aws/aws-cdk#20319]
### Describe the bug

A recent release of `@types/prettier` (2.6.1) updated the minimum `typescript` version from `3.7` to `4.2`, which is not supported.

`@types/prettier` is a transitive dependency 
```bash
npm explain @types/prettier                                                                                                                                                       
@types/prettier@2.6.1 dev                                                                                                                                                           
node_modules/@types/prettier                                                                                                                                                        
  @types/prettier@"^2.1.5" from jest-snapshot@27.5.1                                                                                                                                
  node_modules/jest-snapshot                                                                                                                                                        
    jest-snapshot@"^27.5.1" from @jest/core@27.5.1                                                                                                                                  
    node_modules/@jest/core                                                                                                                                                         
      @jest/core@"^27.5.1" from jest@27.5.1    
```

### Expected Behavior

No error.

### Current Behavior

When trying to build a typescript app you get the error message: 

```bash
node_modules/@types/prettier/index.d.ts:41:54 - error TS2315: Type 'IsTuple' is not generic.

41 type IndexProperties<T extends { length: number }> = IsTuple<T> extends true
                                                        ~~~~~~~~~~

node_modules/@types/prettier/index.d.ts:53:6 - error TS2456: Type alias 'IsTuple' circularly references itself.

53 type IsTuple<T> = T extends [] ? true : T extends [infer First, ...infer Remain] ? IsTuple<Remain> : false;
        ~~~~~~~

node_modules/@types/prettier/index.d.ts:53:65 - error TS2574: A rest element type must be an array type.

53 type IsTuple<T> = T extends [] ? true : T extends [infer First, ...infer Remain] ? IsTuple<Remain> : false;
                                                                   ~~~~~~~~~~~~~~~

node_modules/@types/prettier/index.d.ts:53:84 - error TS2315: Type 'IsTuple' is not generic.

53 type IsTuple<T> = T extends [] ? true : T extends [infer First, ...infer Remain] ? IsTuple<Remain> : false;
                                                                                      ~~~~~~~~~~~~~~~

node_modules/@types/prettier/index.d.ts:96:5 - error TS2589: Type instantiation is excessively deep and possibly infinite.

96     call<
       ~~~~

node_modules/@types/prettier/index.d.ts:131:5 - error TS2589: Type instantiation is excessively deep and possibly infinite.

131     each<
        ~~~~

node_modules/@types/prettier/index.d.ts:165:5 - error TS2589: Type instantiation is excessively deep and possibly infinite.

165     map<
        ~~~


Found 7 errors.
```

### Reproduction Steps

Step 1 - create a new CDK app

```bash
cdk init --language=typescript app
```

Step 2 - Try to build

```bash
npm run build
```

### Possible Solution

Pin `@types/prettier` to `2.6.0` in `devDependencies`

_package.json_
```
{
  "devDependencies": {
    "@types/prettier": "2.6.0"
  }
}
```

### CDK CLI Version

2.24.0

### Node.js Version

16.14.0

### OS

Linux

### Language

Typescript

**Comment by corymhall** [comment]
Issue in DefinitelyTyped
https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/60310

**Comment by biffgaut** [comment]
We're seeing it too.

**Comment by biffgaut** [comment]
Any build of a brand new CDK program is breaking:

```
AWS/Constructs/clients
▶ md pretty

AWS/Constructs/clients
▶ cd pretty

Constructs/clients/pretty
▶ cdk init --language typescript

Applying project template app for typescript
# Welcome to your CDK TypeScript project

This is a blank project for TypeScript development with CDK.

The `cdk.json` file tells the CDK Toolkit how to execute your app.

## Useful commands

* `npm run build`   compile typescript to js
* `npm run watch`   watch for changes and compile
* `npm run test`    perform the jest unit tests
* `cdk deploy`      deploy this stack to your default AWS account/region
* `cdk diff`        compare deployed stack with current state
* `cdk synth`       emits the synthesized CloudFormation template

Initializing a new git repository...
bin/pretty.ts:18:  // env: { account: '123456789012', region: 'us-east-1' },

[ERROR] Matched one or more prohibited patterns

Possible mitigations:
- Mark false positives as allowed using: git config --add secrets.allowed ...
- Mark false positives as allowed by adding regular expressions to .gitallowed at repository's root directory
- List your configured patterns: git config --get-all secrets.patterns
- List your configured allowed patterns: git config --get-all secrets.allowed
- List your configured allowed patterns in .gitallowed at repository's root directory
- Use --no-verify if this is a one-time false positive
Unable to initialize git repository for your project.
Executing npm install...
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm notice
npm notice New minor version of npm available! 8.3.0 -> 8.10.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v8.10.0
npm notice Run npm install -g npm@8.10.0 to update!
npm notice
✅ All done!


Constructs/clients/pretty  master ✗                                                                                         ✚ ◒
▶ npm run build

> pretty@0.1.0 build
> tsc

node_modules/@types/prettier/index.d.ts:41:54 - error TS2315: Type 'IsTuple' is not generic.

41 type IndexProperties<T extends { length: number }> = IsTuple<T> extends true
                                                        ~~~~~~~~~~

node_modules/@types/prettier/index.d.ts:53:6 - error TS2456: Type alias 'IsTuple' circularly references itself.

53 type IsTuple<T> = T extends [] ? true : T extends [infer First, ...infer Remain] ? IsTuple<Remain> : false;
        ~~~~~~~

node_modules/@types/prettier/index.d.ts:53:65 - error TS2574: A rest element type must be an array type.

53 type IsTuple<T> = T extends [] ? true : T extends [infer First, ...infer Remain] ? IsTuple<Remain> : false;
                                                                   ~~~~~~~~~~~~~~~

node_modules/@types/prettier/index.d.ts:53:84 - error TS2315: Type 'IsTuple' is not generic.

53 type IsTuple<T> = T extends [] ? true : T extends [infer First, ...infer Remain] ? IsTuple<Remain> : false;
                                                                                      ~~~~~~~~~~~~~~~

node_modules/@types/prettier/index.d.ts:96:5 - error TS2589: Type instantiation is excessively deep and possibly infinite.

96     call<
       ~~~~

node_modules/@types/prettier/index.d.ts:131:5 - error TS2589: Type instantiation is excessively deep and possibly infinite.

131     each<
        ~~~~

node_modules/@types/prettier/index.d.ts:165:5 - error TS2589: Type instantiation is excessively deep and possibly infinite.

165     map<
        ~~~


Found 7 errors.

npm notice
npm notice New minor version of npm available! 8.3.0 -> 8.10.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v8.10.0
npm notice Run npm install -g npm@8.10.0 to update!
npm notice

Constructs/clients/pretty  master ✗                                                                                        ✚ ◒  ⍉
▶
```

Here is the package.json for the project:

```
{
  "name": "pretty",
  "version": "0.1.0",
  "bin": {
    "pretty": "bin/pretty.js"
  },
  "scripts": {
    "build": "tsc",
    "watch": "tsc -w",
    "test": "jest",
    "cdk": "cdk"
  },
  "devDependencies": {
    "@types/jest": "^26.0.10",
    "@types/node": "10.17.27",
    "jest": "^26.4.2",
    "ts-jest": "^26.2.0",
    "aws-cdk": "2.20.0",
    "ts-node": "^9.0.0",
    "typescript": "~3.9.7"
  },
  "dependencies": {
    "aws-cdk-lib": "2.20.0",
    "constructs": "^10.0.0",
    "source-map-support": "^0.5.16"
  }
}
```

```
Constructs/clients/pretty  master ✗                                                                                        ✚ ◒  ⍉
▶ cdk --version
2.20.0 (build 738ef49)
```

**Comment by johnnyshankman** [comment]
+1 Had the same issue this morning. 

Solved it by pinning the version to `2.6.0` in my `devDependencies` and suddenly no issues.

```
"devDependencies": {
    "@aws-cdk/assertions": "1.143.0",
    "@types/jest": "^26.0.10",
    "@types/node": "10.17.27",
    "aws-cdk": "1.143.0",
    "ts-node": "^9.0.0",
    "typescript": "~3.9.7",
    "@types/prettier": "2.6.0"
  },
```
